### ISSUE TITLE
Comparison with EOF always false on Termux platform

### ISSUE THREAD

**Issue by PhilipRoman** [PhilipRoman/hexproc#15]
# Description

When compiling Hexproc on a Termux platform, there is a warning:
```
$ make
c99 -std=c99 -pedantic -pipe  -DHEXPROC_DATE="\"2020-02-19 08:17:14+00:00\""  -DHEXPROC_VERSION="\"v0.2.6\"" -O2 -s -c -o build/linux/hexproc.o hexproc.c
clang-9: warning: argument unused during compilation: '-s' [-Wunused-command-line-argument]
hexproc.c:141:49: warning: result of comparison of
      constant -1 with expression of type 'char'
      is always true
      [-Wtautological-constant-out-of-range-compare]
  ...(byte = bytequeue_get(&buffer)) != EOF;)
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~
1 warning generated.
```
Running the executable creates an infinite stream of 0xFF octets.

This is a regression as I was previously able to compile and run Hexproc on this platform.

## Software versions

Git tag `v0.2.6`
```
$ uname -a
Linux localhost 3.18.35 #1 SMP PREEMPT Fri Nov 30 13:25:19 IST 2018 armv7l Android
```

**Comment by PhilipRoman** [comment]
Should be fixed with 549e2d40cbc8ee43ca033c7d0791e9b2bbaefda6, requires testing.

**Comment by PhilipRoman** [comment]
Confirmed fix, released v0.2.7
